### PR TITLE
Create org with same name as user on account creation

### PR DESCRIPTION
**Pull request by jlmcgehee21**
Why:

* https://github.com/nerves-hub/nerves_hub_web/issues/168
* This is the next step to moving to the github org.

This change addresses the need by:

* Cleaning up some of the user changeset logic.
* Removing the `org_name` field from the account creation form.
* Injecting `org_name` into the user params in the controller.

Other changes:

* There was a "bug" where the password reset token was not expired when
it was used, so that was fixed and tested.

### CODE REVIEW

**Comment by mobileoverlord** [comment]
One thing we have to consider as part of this is how to migrate the data that is already in production. Currently there is only one org `nerves-hub` but it has a couple users including the default user. We should probably delete the users and re create them. This should probably be done as a migration so it only happens once. Then we need to update the `seeds.exs`

**Comment by jlmcgehee21** [comment]
hmm yeah. I can add that to the migration.

**Comment by jlmcgehee21** [comment]
@mobileoverlord I'm definitely not an expert at data migrations in Ecto, and am not super in love with what I threw together, and feedback or total re-work is welcome.

File: apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        o in Org,
+        where: o.id not in ^new_org_ids,
+        preload: :users
+      )
+      |> Repo.all()
+      |> Enum.filter(fn o -> Enum.empty?(o.users) end)
+      |> Enum.map(fn o -> Repo.delete(o) end)
+  end
+
+  def down do
```
Comment (mobileoverlord) [inline apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs]: There is no going back :)

Comment (jlmcgehee21) [reply apps/nerves_hub_core/priv/repo/migrations/20180810151416_default_user_org.exs]: Lol, yeah. Don't think there's a way to make that happen

File: apps/nerves_hub_core/lib/nerves_hub_core/accounts.ex [hunk 2]
```diff
@@ -89,7 +100,7 @@ defmodule NervesHubCore.Accounts do
   def create_user(%Org{} = org, params) do
```
Comment (mobileoverlord) [inline apps/nerves_hub_core/lib/nerves_hub_core/accounts.ex]: We should make it that create_user essentially does create_user_with_org with the params set that it tries to create the default org. You should never be able to create a new user without creating an org by the same name

**Comment by mobileoverlord** [comment]
I cleaned up a warning and rebased this on master. Make sure to pull

File: apps/nerves_hub_core/lib/nerves_hub_core/accounts.ex (hunk 2 above)
Comment (jlmcgehee21) [reply apps/nerves_hub_core/lib/nerves_hub_core/accounts.ex]: @mobileoverlord I agree, but was going to save this (and some other desirable refactors) for when I added the `many-many` relationship between users/orgs.

**Comment by mobileoverlord** [comment]
🚀 Lets do this!

**Pull request merged**